sdks/java/core/src/main/java/org/apache/beam/sdk/testing/FileChecksumMatcher.java

```diff
@@ -66,48 +99,105 @@ public FileChecksumMatcher(String checksum, String filePath) {
 
     this.expectedChecksum = checksum;
     this.filePath = filePath;
+    this.shardTemplate =
+        Pattern.compile(shardTemplate == null ? DEFAULT_SHARD_TEMPLATE : shardTemplate);
```

Comment:
This shouldn't check for `null` but should require the user to call the other constructor.